src/hooks/useSearchCurrencyList.ts

```diff
@@ -1,473 +1,243 @@
+
+  const selectTopSearchResults = useCallback(
+    (data: TokenSearchResult) => {
+      const results = data.filter(asset => {
+        const isFavorite = favoriteAddresses.map(a => a?.toLowerCase()).includes(asset.uniqueId?.toLowerCase());
+        if (isFavorite) return false;
 
-  const getImportedAsset = useCallback(
-    async (searchQuery: string, chainId: number): Promise<RainbowToken[] | null> => {
-      if (searching) {
-        if (isAddress(searchQuery)) {
-          const tokenListEntry = rainbowTokenList.RAINBOW_TOKEN_LIST[searchQuery.toLowerCase()];
-          if (tokenListEntry) {
-            return [tokenListEntry];
-          }
-          const provider = getProvider({ chainId });
-          const tokenContract = new Contract(searchQuery, erc20ABI, provider);
-          try {
-            const [name, symbol, decimals, address] = await Promise.all([
-              tokenContract.name(),
-              tokenContract.symbol(),
-              tokenContract.decimals(),
-              getAddress(searchQuery),
-            ]);
-            const uniqueId = getUniqueId(address, chainId);
+        const hasIcon = asset.icon_url;
+        const isMatch = hasIcon || searchQuery.length > 2;
 
-            return [
-              {
-                chainId,
-                address,
-                decimals,
-                favorite: false,
-                highLiquidity: false,
-                isRainbowCurated: false,
-                isVerified: false,
-                name,
-                networks: {
-                  [chainId]: {
-                    address,
-                    decimals,
-                  },
-                },
-                symbol,
-                network: useBackendNetworksStore.getState().getChainsName()[chainId],
-                uniqueId,
-              } as RainbowToken,
-            ];
-          } catch (e) {
-            logger.warn('[useSearchCurrencyList]: error getting token data', { error: (e as Error).message });
-            return null;
-          }
+        if (!isMatch) {
+          const crosschainMatch = getExactMatches([asset], searchQuery);
+          return crosschainMatch.length > 0;
         }
-      }
-      return null;
-    },
-    [searching]
-  );
 
-  const getCrosschainVerifiedAssetsForNetwork = useCallback(
-    async (chainId: ChainId) => {
-      const results = await searchCurrencyList({
-        searchList: 'verifiedAssets',
-        query: '',
-        chainId,
+        return isMatch;
       });
-      setCrosschainVerifiedAssets(state => ({
-        ...state,
-        [chainId]: handleSearchResponse(results || []),
-      }));
+      const topResults = searchQuery === '' ? sortForDefaultList(results) : sortTokensByRelevance(results, searchQuery);
+      return topResults.slice(0, MAX_VERIFIED_RESULTS);
     },
-    [handleSearchResponse]
+    [searchQuery, favoriteAddresses]
   );
 
-  const getCrosschainVerifiedAssets = useCallback(async () => {
```

Comment:
a separate call for crosschain assets is no longer needed, as the new v3 endpoint will look for the most relevant token across chains